kuma/users/stripe_utils.py

```diff
@@ -101,8 +102,13 @@ def cancel_stripe_customer_subscriptions(user):
     customer = stripe.Customer.retrieve(user.stripe_customer_id)
     for sub in customer.subscriptions.data:
         s = stripe.Subscription.retrieve(sub.id)
-        UserSubscription.set_canceled(user, s.id)
-        s.delete()
+        try:
+            UserSubscription.set_canceled(user, s.id)
+            s.delete()
```

Comment:
First of all, since this isn't wrapped in an atomic transaction what's going to happen is that `UserSubscription.set_canceled(user, s.id)` is called and fine but the next line doesn't actually do anything. 
The simplest solution, from a quick skim, is to move `UserSubscription.set_canceled(user, s.id)` to be right before the `return True`